frontend/src/utils/CampUtils.ts

```diff
@@ -67,22 +83,28 @@ export const sortScheduledSessions = (
   return sessions.sort((a, b) => a.startDate.getTime() - b.startDate.getTime());
 };
 
-export const getSessionDatesRangeString = (
-  session: CreateCampSession,
-): string => {
-  const startDateOptions: Intl.DateTimeFormatOptions = {
-    month: "short",
-    day: "numeric",
-  };
-  const endDateOptions: Intl.DateTimeFormatOptions = {
-    month: "short",
-    day: "numeric",
-    year: "numeric",
-  };
-  return `${session.startDate.toLocaleDateString(
-    "en-US",
-    startDateOptions,
-  )} - ${session.endDate.toLocaleDateString("en-US", endDateOptions)}`;
+export const getSessionDates = (
+  sessionStartDate: Date,
+  selectedWeekDayValues: boolean[],
+): Date[] => {
+  const dates: Date[] = [];
+
+  let currDay = sessionStartDate.getDay();
+  for (
+    let daysAfterStartDate = 0;
+    daysAfterStartDate < 7;
+    daysAfterStartDate += 1
+  ) {
+    // only add days that the user selected
+    // e.g. only add Mondays - Fridays dates, don't add weekends
+    if (selectedWeekDayValues[currDay]) {
+      const newDate = new Date(sessionStartDate.getTime());
```

Comment:
nit: IIRC this deep copies a date, right? I'm wondering if it's worth creating some utility method on the Date prototype (a la https://github.com/Microsoft/TypeScript/issues/7726#issuecomment-234621084) so we could just do `sessionStartDate.deepCopy()`

might not be worth it since we'd have to mess with built-in JS objects - could you just leave a quick note that this is deep copying the Date object?